pkg/workloads/cortex/lib/context.py

```diff
@@ -168,11 +168,26 @@ def get_predictor_impl(self, api_name, project_dir):
             refresh_logger()
 
         try:
-            _validate_impl(impl, PREDICTOR_IMPL_VALIDATION)
+            classes = inspect.getmembers(impl, inspect.isclass)
+            if len(classes) == 0:
+                raise UserException("no class definitions found")
```

Comment:
nit: `Predictor class is not defined` (or if you are ok with this message, you could delete this check since the same message will be raised later